baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/AvoidNewHashMapInt.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.baseline.errorprone;
+
+import com.google.auto.service.AutoService;
+import com.google.common.collect.Maps;
+import com.google.errorprone.BugPattern;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.bugpatterns.BugChecker;
+import com.google.errorprone.fixes.SuggestedFix;
+import com.google.errorprone.fixes.SuggestedFixes;
+import com.google.errorprone.matchers.Description;
+import com.google.errorprone.matchers.Matcher;
+import com.google.errorprone.matchers.method.MethodMatchers;
+import com.sun.source.tree.ExpressionTree;
+import com.sun.source.tree.NewClassTree;
+
+@AutoService(BugChecker.class)
+@BugPattern(
+        name = "AvoidNewHashMapInt",
+        link = "https://github.com/palantir/gradle-baseline#baseline-error-prone-checks", // TODO (DCohen)
+        linkType = BugPattern.LinkType.CUSTOM,
+        severity = BugPattern.SeverityLevel.WARNING,
+        summary = "The HashMap(int) constructor is misleading, use Maps.newHashMapWithExpectedSize(int) instead.")
+public final class AvoidNewHashMapInt extends BugChecker implements BugChecker.NewClassTreeMatcher {
+
+    private static final long serialVersionUID = 1L;
+
+
+    private static final Matcher<ExpressionTree> NEW_HASH_MAP = MethodMatchers.constructor()
+            .forClass("java.util.HashMap")
+            .withParameters("int");
+
+    @Override
+    public Description matchNewClass(
+            NewClassTree tree, VisitorState state) {
+        if (!NEW_HASH_MAP.matches(tree, state)) {
+            return Description.NO_MATCH;
+        }
+
+        SuggestedFix.Builder fixBuilder = SuggestedFix.builder();
+        String newType = SuggestedFixes.qualifyType(state, fixBuilder, Maps.class.getName());
+        String arg = tree.getArguments().isEmpty()
```

Comment:
isEmpty should never evaluate to true for a constructor which takes an `int` argument --  we can either throw, or return `Description.NO_MATCH`. I tend to prefer the latter because we don't want to fail build sin cases that we haven't considered or don't understand.